### PR TITLE
Show booked dates

### PR DESCRIPTION
**Pull request by ovan**

### CODE REVIEW

File: app/views/listings/_listing_actions.haml [hunk 1]
```diff
@@ -23,61 +23,108 @@
...
+              %input#booking-start-output{:type => 'hidden', :name => 'start_on'}
+
+            .datepicker-end-wrapper
+              = label_tag t(".booking_to")
+              %input.input-sm.form-control.required#end-on{:type => 'text', :name => "end_on", :placeholder => t("datepicker.format"), :data => { :output => "booking-end-output" }}
+              %input#booking-end-output{:type => 'hidden', :name => 'end_on'}
+        - blocked_dates_result.on_error do
+
+          .listings-availability-information-error
+            = "Could not load availabilty information. Please try again later."
```
Comment (ovan) [inline app/views/listings/_listing_actions.haml]: This error str is not hard coded. I guess there's no reason to have it localized?

File: app/views/listings/_listing_actions.haml [hunk 2]
```diff
@@ -23,61 +23,108 @@
...
+          - translated_days_min = days.map { |day_symbol| t("datepicker.days_min.#{day_symbol}") }.to_json
+          - translated_months = months.map { |day_symbol| t("datepicker.months.#{day_symbol}") }.to_json
+          - translated_months_short = months.map { |day_symbol| t("datepicker.months_short.#{day_symbol}") }.to_json
+
+          - content_for :extra_javascript do
+            :javascript
+
+              var disabledDates = [
+                // TODO Server-render disabled dates here
+              ]
```
Comment (ovan) [inline app/views/listings/_listing_actions.haml]: Is this deprecated test code or something missing from repo?

File: app/controllers/listings_controller.rb [hunk 3]
```diff
@@ -543,6 +556,38 @@ def create_bookable(community_uuid, listing_uuid, author_uuid, auth_context)
     end
   end
 
+  def get_blocked_dates(community, user, listing)
+    start_on = Date.today - 1.day
```
Comment (rap1ds) [inline app/controllers/listings_controller.rb]: The Railsy way is to say `1.day.ago`

File: app/controllers/listings_controller.rb [hunk 4]
```diff
@@ -543,6 +556,38 @@ def create_bookable(community_uuid, listing_uuid, author_uuid, auth_context)
...
+    end_at = Date.today + 12.months
```
Comment (rap1ds) [inline app/controllers/listings_controller.rb]: Again, Railsy way: `12.months.from_now`

File: app/assets/javascripts/datepicker.js [hunk 5]
```diff
@@ -22,6 +36,14 @@ window.ST = window.ST || {};
...
     var picker = dateRage.datepicker(options);
 
+    $("#start-on").focus(function() {
+      $("#start-on").datepicker("setDatesDisabled", disabledStartDates);
+    });
+
+    $("#end-on").focus(function() {
+      $("#end-on").datepicker("setDatesDisabled", disabledEndDates);
+    });
+
```
Comment (kpuputti) [inline app/assets/javascripts/datepicker.js]: ❤️ 😍 🎉 🍬

**Pull request merged**